===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(cleanup_directory:remove->buckets)) at 44 (line 561, column 7),
        Insert: if (true) free(*(cleanup_directory:remove)) at 46 (line 562, column 4) }
===== Actual patch in diff format =====
@@ -559,6 +559,8 @@ cleanup_directory (const char *directory,
 			continue;
 
 		if (asprintf (&path, "%s/%s", directory, dp->d_name) < 0)
+			free(remove->buckets);
+			free(remove);
 			return_val_if_reached (false);
 
 
